Bitwig Studio 4.0.1: can't open gtk file dialog

### Describe the bug
When I try to open a file dialog in Bitwig Studio 4.1.0 I am getting that error:
```
This command is not for general use and should only be run as the result of a call to
ProcessBuilder.start() or Runtime.exec() in a java application
[2021-08-14 16:34:17.437 x11-windowing-system error] Error launching show file dialog command:
	com.bitwig.base.io.ProcessException: Error retrieving localized message for exception class com.bitwig.base.io.ProcessException java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key ProcessException.message
	at com.bitwig.base.io.RKn.Fep(SourceFile:412)
	at com.bitwig.base.io.RKn.Fep(SourceFile:105)
	at com.bitwig.x11_windowing_system.X11FileDialog.OgE(SourceFile:206)
	at com.bitwig.x11_windowing_system.X11FileDialog.bWD(SourceFile:32)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.IOException: Cannot run program "/nix/store/vwhrcd9jiy0gc3jb2b60v4p02qibikdy-bitwig-studio-4.0.1/libexec/bin/show-file-dialog-gtk3": error=0, Failed to exec spawn helper: pid: 617783, exit value: 1
	at java.base/java.lang.ProcessBuilder.start(Unknown Source)
	at java.base/java.lang.ProcessBuilder.start(Unknown Source)
	at com.bitwig.base.io.RKn.Fep(SourceFile:129)
	... 4 more
Caused by: java.io.IOException: error=0, Failed to exec spawn helper: pid: 617783, exit value: 1
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(Unknown Source)
	at java.base/java.lang.ProcessImpl.start(Unknown Source)
	... 7 more
```
I can manually start `/nix/store/vwhrcd9jiy0gc3jb2b60v4p02qibikdy-bitwig-studio-4.0.1/libexec/bin/show-file-dialog-gtk3` from command line and I'm seeing the file dialog and it returns the selected file.

Unsure if this is of any use but [here](https://gist.github.com/landorg/671d0920dfb6ff623def3d060ad21d66) is the decompiled class that calls `show-file-dialog-gtk3`.

### Steps To Reproduce
Steps to reproduce the behavior:
1. open bitwig
2. click `Offline?`
3. click `Load Activation File` (or open any other file dialog in bitwig)

### Expected behavior
GTK file dialog opens

### Notify maintainers
@SuperSandro2000 @Isti115 @r-ryantm @bfortz  @michalrus  @mrVanDalo 

### Metadata
```console
[I] ➜ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/wzach25vwapl3sa6f7gylqank3jhhsin-bash-interactive-4.4-p23-dev
copying path '/nix/store/wzach25vwapl3sa6f7gylqank3jhhsin-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.84-rt47, NixOS, 21.05.2428.2d6ab6c6b92 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.05.2428.2d6ab6c6b92, nixos-unstable-21.11pre307912.fe01052444c"`
 - channels(roland): `"home-manager-21.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixos modules affected by the problem
module: bitwig-studio
```

**EDIT:** added more information

I tried to do the same thing and I noticed that the exact error message changed from:
```
Caused by: java.io.IOException: Cannot run program "/nix/store/g8iwg2v0llmlnrha5j51kc8frfbrlwgs-bitwig-studio-4.0.1/libexec/bin/show-file-dialog-gtk3": error=0, Failed to exec spawn helper: pid: 22579, exit value: 1
```
to
```
Caused by: java.io.IOException: Cannot run program "/nix/store/x3w6048pvcpqqyifgrh00l0cdqvxm1px-bitwig-studio-4.0.1/libexec/bin/show-file-dialog-gtk3": error=2, No such file or directory
```
but I can still run `/nix/store/x3w6048pvcpqqyifgrh00l0cdqvxm1px-bitwig-studio-4.0.1/libexec/bin/show-file-dialog-gtk3`.
Not sure what that means

--- a/pkgs/applications/audio/bitwig-studio/bitwig-studio4.nix
+++ b/pkgs/applications/audio/bitwig-studio/bitwig-studio4.nix
@@ -49,6 +49,7 @@ stdenv.mkDerivation rec {
       -not -name '*.so.*' \
       -not -name '*.so' \
       -not -name '*.jar' \
+      -not -name 'jspawnhelper' \
       -not -path '*/resources/*' | \
     while IFS= read -r f ; do
       patchelf --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" $f
@@ -58,6 +59,10 @@ stdenv.mkDerivation rec {
         --suffix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath buildInputs}"
     done
 
+    find $out -type f -executable -name 'jspawnhelper' | \
+    while IFS= read -r f ; do
+      patchelf --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" $f
+    done
   '';
 
   meta = with lib; {

